src/util/stream.rs

```diff
@@ -19,6 +19,29 @@ use std::time::Duration;
 ///
 /// [`timeout`]: #method.timeout
 pub trait StreamExt: Stream {
+    /// Throttle down the stream by enforcing a fixed delay between items.
+    ///
+    /// Errors are also delayed.
+    fn throttle(self, duration: Duration) -> Throttle<Self>
+    where Self: Sized
+    {
+        Throttle::new(self, duration)
+    }
+
+    /// Throttle down the stream by enforcing that a maximum of `rate_per_sec`
+    /// items will be passed through per second.
+    ///
+    /// Errors are also delayed.
+    fn rate_limit(self, max_per_sec: f64) -> Throttle<Self>
```

Comment:
This isn't really a rate limit implementation as it could prevent `max_per_sec` from actually being passed through. I'd probably leave this function out and stick with only `throttle` for now.